Add notice about plan credits to /plans page to reduce users confusion about pricing

adamziel opened the pull request:
Related to p9jf6J-So-p2

This PR adds some messaging on `/plans` page that better explains the nature of discounted prices.

An example:

<img width="1088" alt="zrzut ekranu 2018-08-31 o 15 31 51" src="https://user-images.githubusercontent.com/205419/44915331-ff6b6b00-ad32-11e8-93f7-d6c9b8befcca.png">

Some specifics are still being discussed so no description or test plan for now.

mattwiebe commented:
This still has some instances where the banner label is getting bumped onto the next line. It's a hard thing to place and we probably need to get some design help on this.

adamziel commented:
@mattwiebe it is getting bumped, I just made the margins right to make it look more natural. It's a tricky problem that we may not be able to avoid, see the discussion under p9jf6J-So-p2. If you have any ideas how to solve it in a different way feel free to add to this PR!

mattwiebe commented on client/my-sites/plan-features/index.jsx:
```diff
@@ -139,6 +140,69 @@ class PlanFeatures extends Component {
+			>
+				{ translate(
+					'You have {{b}}%(credits)s{{/b}} in upgrade credits available! ' +
+						'Apply the value of your current plan towards an upgrade before your credits expire!',
```
I think I would get rid of the double exclamation marks here, a little too intense. Keep the first one and change the second to just a period, I'd say.

jancavan commented:
@adamziel I just added credits to my account, but I'm unable to get the banner to show up in `/plans`. Do you have steps on how to repro?

adamziel commented:
@jancavan I updated the color, thanks!

I know the word "credits" may sound like free/paid credits, but it actually means the amount you already paid for your current plan. There are strong reasons not to say "discount" so we chose this word, there are links to more context in the P2 post linked to in the description.

About using the `<Badge/>` component, it is a good suggestion I tried that. It doesn't help much though because this specific use case requires a "compact" badge so we had to customize most CSS rules anyway, and also any future change to Badge would affect this screen in an unexpected way. See here:

With current CSS:

<img width="341" alt="zrzut ekranu 2018-09-07 o 11 28 21" src="https://user-images.githubusercontent.com/205419/45214219-24fbf580-b29a-11e8-81ed-192364b06eb6.png">

With <Badge/> component:

<img width="338" alt="zrzut ekranu 2018-09-07 o 11 28 33" src="https://user-images.githubusercontent.com/205419/45214216-24635f00-b29a-11e8-9492-9949c40acf54.png">

So I'm going to merge this without the <Badge />.

@mattwiebe good spot! I updated the copy.

adamziel commented:
@jancavan Actually when I think about it, I may just create a `.is-compact` version of badge component in a subsequent PR, this would be the best solution here.

jancavan commented:
@adamziel Sounds good. Yeah, the badge component looks better. Even if we added custom styles to it, we still save a couple lines of css.

ockham commented on client/my-sites/plan-features/index.jsx:
```diff
@@ -139,6 +141,69 @@ class PlanFeatures extends Component {
+			return null;
+		}
+
+		return ReactDOM.createPortal(
```
:weary: Please no Portals -- using them to just 'get a component in place' somewhere, they are the UI equivalent of :spaghetti: code. See prior discussion [here](https://github.com/Automattic/wp-calypso/pull/22384#issuecomment-382773722).

adamziel replied:
@ockham interesting! Portals are already used to display another notice from the same component and I simply copied that. I don't agree we should avoid Portals for all cost, although in case of this component we could totally rid of them in a short janitorial PR.
